### PR TITLE
AP_VideoTX: Change videoTX to optional

### PR DESCRIPTION
**Pull request by muramura**
I would make VideoTX an option.
I don't have the video capability.
I want to be able to select the function.

### CODE REVIEW

**Comment by andyp1per** [comment]
Its deliberately like this because the code is small and it makes the code more complicated making it an option

**Comment by peterbarker** [comment]
Not a small saving, but the default would probably have to change as it is relevant on a lot of small boards.

```
Binary           text              data         bss            total
---------------  ----------------  -----------  -------------  ----------------
ardusub          -664 (-0.0767%)   0 (0.0000%)  0 (0.0000%)    -664 (-0.0625%)
arducopter       -1052 (-0.1095%)  0 (0.0000%)  4 (+0.0020%)   -1048 (-0.0905%)
ardurover        -964 (-0.1084%)   0 (0.0000%)  -4 (-0.0020%)  -968 (-0.0891%)
arduplane        -944 (-0.0967%)   0 (0.0000%)  0 (0.0000%)    -944 (-0.0805%)
arducopter-heli  -1028 (-0.1087%)  0 (0.0000%)  4 (+0.0020%)   -1024 (-0.0896%)
antennatracker   -680 (-0.0927%)   0 (0.0000%)  0 (0.0000%)    -680 (-0.0731%)
```

**Comment by auturgy** [comment]
A good example for pushing forward with a configurable online build server.

Regards,

James

File: libraries/AP_VideoTX/AP_VideoTX.h [hunk 1]
```diff
@@ -15,6 +15,13 @@
 #pragma once
 
 #include <AP_HAL/AP_HAL.h>
+
+#ifndef HAL_VIDEOTX_ENABLED
+#define HAL_VIDEOTX_ENABLED !HAL_MINIMIZE_FEATURES
```
Comment (tridge) [inline libraries/AP_VideoTX/AP_VideoTX.h:20]: should default enabled so we don't break existing setups

**Review by tridge — CHANGES_REQUESTED** [review]

**Comment by peterbarker** [comment]
@muramura needs to pass CI.

Default needs changing.  You should also state that you've tested it with the `HAL_*_ENABLED` feature turned both on and off.

File: libraries/AP_VideoTX/AP_VideoTX.h (hunk 1 above)
Comment (andyp1per) [reply libraries/AP_VideoTX/AP_VideoTX.h:20]: The other option would be to make it dependent on SmartAudio and CRSF but default on is simpler and my preference

**Comment by peterbarker** [comment]
This is now conflicting horribly, and review comments haven't been addressed.  Please reopen this if you feel like chasing it.

**Pull request closed without merging**